### PR TITLE
[server] Update OpenAI endpoints

### PR DESCRIPTION
**Pull request by dsikka**
# Summary
- Add the `/v1/completions` endpoint
- Update `/v1/chat/completions` to accept/handle FastChat-compliant dictionaries 

# Testing

- Using OpenAI API:

```python
from openai import OpenAI


client = OpenAI(base_url="http://localhost:5543/v1", api_key="EMPTY")

models = client.models.list()

model = "hf:neuralmagic/mpt-7b-chat-pruned50-quant"
print(f"Accessing model API '{model}'")


# Completion API
stream = True
completion = client.chat.completions.create(
    messages={"role": "user", "content": "Talk about the Toronto Raptors."},
    stream=stream,
    max_tokens=100,
    model=model,
)

print("Chat results:")
if stream:
    text = ""
    for c in completion:
        print(c)
else:
    print(completion)


stream = True
completion = client.completions.create(
    prompt="How are you today?",
    stream=stream,
    max_tokens=100,
    model=model,
)

print("Completion results:")
if stream:
    text = ""
    for c in completion:
        print(c)
else:
    print(completion)
```

### CODE REVIEW

File: src/deepsparse/server/openai_server.py
Comment (dbogunowicz) [inline src/deepsparse/server/openai_server.py:223]: nit: i am afraid that `request` can be a beefy object and printing it to terminal may not be aesthetically pleasing

**Review by dbogunowicz — APPROVED** [review]

File: src/deepsparse/server/openai_server.py
Comment (bfineran) [inline src/deepsparse/server/openai_server.py:107]: let's push this error to the client side instead of server side

File: src/deepsparse/server/openai_server.py
Comment (bfineran) [inline src/deepsparse/server/openai_server.py:123]: same here

**Comment by mgoin** [comment]
I ran through the script using `hf:neuralmagic/TinyLlama-1.1B-Chat-v0.4-pruned50-quant-ds` as the model and installing `pip install fschat accelerate`

Looks like something about the last message handshake went wrong.

[client.txt](https://github.com/neuralmagic/deepsparse/files/13591881/client.txt)
```
ChatCompletionChunk(id='cmpl-c735b32f15c043b49893cd6a0ac7ab96', choices=[Choice(delta=ChoiceDelta(content='', function_call=None, role=None, tool_calls=None), finish_reason='length', index=0)], created=1701898636, model='hf:neuralmagic/TinyLlama-1.1B-Chat-v0.4-pruned50-quant-ds', object='chat.completion.chunk', system_fingerprint=None)
httpx.RemoteProtocolError: peer closed connection without sending complete message body (incomplete chunked read)
```

[server.txt](https://github.com/neuralmagic/deepsparse/files/13591879/server.txt)
```
  File "/Users/mgoin/code/deepsparse/src/deepsparse/server/openai_server.py", line 159, in abort_request
    await pipeline.abort(request_id)
AttributeError: 'TextGenerationPipeline' object has no attribute 'abort'
```

**Comment by dsikka** [comment]
> I ran through the script using `hf:neuralmagic/TinyLlama-1.1B-Chat-v0.4-pruned50-quant-ds` as the model and installing `pip install fschat accelerate`
> 
> Looks like something about the last message handshake went wrong.
> 
> [client.txt](https://github.com/neuralmagic/deepsparse/files/13591881/client.txt)
> 
> ```
> ChatCompletionChunk(id='cmpl-c735b32f15c043b49893cd6a0ac7ab96', choices=[Choice(delta=ChoiceDelta(content='', function_call=None, role=None, tool_calls=None), finish_reason='length', index=0)], created=1701898636, model='hf:neuralmagic/TinyLlama-1.1B-Chat-v0.4-pruned50-quant-ds', object='chat.completion.chunk', system_fingerprint=None)
> httpx.RemoteProtocolError: peer closed connection without sending complete message body (incomplete chunked read)
> ```
> 
> [server.txt](https://github.com/neuralmagic/deepsparse/files/13591879/server.txt)
> 
> ```
>   File "/Users/mgoin/code/deepsparse/src/deepsparse/server/openai_server.py", line 159, in abort_request
>     await pipeline.abort(request_id)
> AttributeError: 'TextGenerationPipeline' object has no attribute 'abort'
> ```

What script did you use? The example script in the PR description?

File: src/deepsparse/server/openai_server.py
Comment (dsikka) [reply src/deepsparse/server/openai_server.py:238]: yup, adding this to the server ticket already created with v2. will update once migration lands.
https://app.asana.com/0/1201735099598270/1205453345266083/f

**Comment by dsikka** [comment]
Going to hold off on merging this in until I add in a few tests.

File: src/deepsparse/server/openai_server.py
Comment (rahul-tuli) [inline src/deepsparse/server/openai_server.py:222]: Server could receive a multitude of requests, would logging the request each time generate a sea of logs?

File: src/deepsparse/server/openai_server.py
Comment (dsikka) [reply src/deepsparse/server/openai_server.py:434]: ah I was debugging and changed this. good catch.

File: src/deepsparse/server/openai_server.py
Comment (dsikka) [reply src/deepsparse/server/openai_server.py:222]: yes. will remove

File: src/deepsparse/server/openai_server.py
Comment (rahul-tuli) [inline src/deepsparse/server/openai_server.py:93]: nit: use % style formatting for debug messages

File: src/deepsparse/server/openai_server.py
Comment (rahul-tuli) [inline src/deepsparse/server/openai_server.py:259]: Does the pipeline object have a abort method? Isn't it a property of DeepSparseOpenAIEngine

File: src/deepsparse/server/openai_server.py
Comment (rahul-tuli) [inline src/deepsparse/server/openai_server.py:469]: nit: add docstring?

**Review by rahul-tuli — DISMISSED** [review]
LGTM!

**Comment by dsikka** [comment]
Although the tests are all green, holding on merging these in as the test are failing locally.

**Review by bfineran — APPROVED** [review]

**Pull request merged**
